js/ng.js

```diff
@@ -162,17 +162,39 @@ addEventListener('deer-updated', event => {
 })
 
 function parseSections() {
-    const canonValue = document.querySelector('input[deer-key="canonicalReference"]')?.value
-    const _document = document.querySelector('input[deer-key="_document"]')
-    const _section = document.querySelector('input[deer-key="_section"]')
-    const _subsection = document.querySelector('input[deer-key="_subsection"]')
-    const elemSet = [_document, _section, _subsection]
-    if (elemSet.includes(null) || !canonValue?.length) { throw new Error(`Missing elements in ${elemSet.join(', ')}`) }
+    // Get the Canonical Reference Locator value
+    const canonValue = document.querySelector('input[deer-key="canonicalReference"]')?.value;
+    const _document = document.querySelector('input[deer-key="_document"]');
+    const _section = document.querySelector('input[deer-key="_section"]');
+    const _subsection = document.querySelector('input[deer-key="_subsection"]');
 
-    const canonSplit = canonValue.split(/[\s\:\.,;\|#§]/)
-    elemSet.forEach((el, index) => el.value = canonSplit[index])
+    // Create an array of the input fields
+    const elemSet = [_document, _section, _subsection];
+
+    // Check if any of the input fields are missing or if the Canonical Reference Locator is empty
+    if (elemSet.includes(null) || !canonValue?.length) {
+        throw new Error(`Missing elements in ${elemSet.join(', ')}`);
+    }
+
+    // Split the Canonical Reference Locator value using a regex pattern
+    const canonSplit = canonValue.split(/[\s\:\.,;\|#§]/);
+
+    // Iterate through the input fields and populate them with corresponding parts of the split value
+    elemSet.forEach((el, index) => {
+        if (index < canonSplit.length) {
+            // Check if the split part is not "undefined" before assignment
+            if (canonSplit[index] !== "undefined") {
```

Comment:
"undefined" may be a string, but it may also just be the javascript primitive `undefined` so make sure to check that as well.